path_login.go

```diff
@@ -0,0 +1,229 @@
+// Copyright © 2019, Oracle and/or its affiliates.
+package ociauth
+
+import (
+	"context"
+	"fmt"
+	log "github.com/hashicorp/go-hclog"
+	"github.com/hashicorp/vault/sdk/framework"
+	"github.com/hashicorp/vault/sdk/logical"
+	"github.com/oracle/oci-go-sdk/common"
+	"github.com/pkg/errors"
+	"net/http"
+	"strings"
+	"unicode"
+)
+
+// These constants store the required http path & method information for validating the signed request
+const (
+	PathVersionBase = "/v1"
+	PathBaseFormat  = "/auth/oci/login/%s"
+	PathLoginMethod = "get"
+)
+
+// Signing Header constants
+const (
+	// HdrRequestTarget represents the special header name used to refer to the HTTP verb and URI in the signature.
+	HdrRequestTarget = `(request-target)`
+)
+
+func pathLogin(b *backend) *framework.Path {
+	return &framework.Path{
+		Pattern: "login/" + framework.GenericNameRegex("role"),
+		Fields: map[string]*framework.FieldSchema{
+			"requestHeaders": {
+				Type:        framework.TypeHeader,
+				Description: `The signed headers of the client`,
+			},
+			"role": {
+				Type:        framework.TypeLowerCaseString,
+				Description: "Name of the role.",
+			},
+		},
+		Callbacks: map[logical.Operation]framework.OperationFunc{
+			logical.UpdateOperation: b.pathLoginUpdate,
+		},
+
+		HelpSynopsis:    pathLoginSyn,
+		HelpDescription: pathLoginDesc,
+	}
+}
+
+func (b *backend) pathLoginUpdate(ctx context.Context, req *logical.Request, data *framework.FieldData) (*logical.Response, error) {
+
+	//Validate the role
+	role, ok := data.GetOk("role")
+	if !ok {
+		return logical.ErrorResponse("Role is not specified"), nil
+	}
+	roleName := role.(string)
+
+	b.Logger().Debug(req.ID, "pathLoginUpdate roleName", roleName)
+
+	//Parse the authentication headers
+	authenticateRequestHeaders := data.Get("requestHeaders").(http.Header)
```

Comment:
If there are no headers, we should return and provide a request error saying the parameter is missing.

Reply:
Done